pyodide-build/pyodide_build/create_pypa_index.py

```diff
@@ -0,0 +1,93 @@
+from pathlib import Path
+from textwrap import dedent
+from typing import TypedDict
+
+
+class PackageInfo(TypedDict):
+    file_name: str
+    sha256: str
+
+
+def create_pypa_index(
+    packages: dict[str, PackageInfo], target_dir: Path, dist_url: str
+) -> None:
+    """Create a pip-compatible Python package (pypa) index to be used with a Pyodide virtual
+    environment.
+
+    To use, pass as an `--index-url` or `--extra-index-url` parameter to pip.
+    The argument should be a `file:` url pointing to the `pypa_index` folder (or
+    if you serve `pypa_index` it can be a normal url). It is also used
+    automatically by Pyodide virtual environments created from a release version
+    of Pyodide.
+
+    Parameters
+    ----------
+    packages:
+        A dictionary of packages that we want to index. This should be the
+        "packages" field from repodata.json.
+
+    target_dir:
+        Where to put the  index. It will be placed in a subfolder of
+        target_dir called `pypa_index`. `target_dir` should exist but
+        `target_dir/pypa_index` should not exist.
```

Comment:
Maybe raise exception when `target_dir` doesn't exist or `target_dir/pypa_index` exists? I mean, `index_dir.mkdir()` will raise exception if it fails, but it would be nice to give a more informative error messages.